plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -663,22 +661,23 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
         last_discovery_at: tx.fn.now(),
       });
 
-      // TODO(Rugvip): only tested towards Postgres and SQLite
+      // TODO(Rugvip): only tested towards MySQL, Postgres and SQLite.
       // We have to do this because the only way to detect if there was a conflict with
       // SQLite is to catch the error, while Postgres needs to ignore the conflict to not
       // break the ongoing transaction.
-      if (!tx.client.config.client.includes('sqlite3')) {
+      if (tx.client.config.client.includes('pg')) {
         query = query.onConflict('entity_ref').ignore() as any; // type here does not match runtime
       }
 
       // Postgres gives as an object with rowCount, SQLite gives us an array
       const result: { rowCount?: number; length?: number } = await query;
       return result.rowCount === 1 || result.length === 1;
     } catch (error) {
-      // SQLite reached this rather than the rowCount check above
+      // SQLite, or MySQL reached this rather than the rowCount check above
       if (
         isError(error) &&
-        error.message.includes('UNIQUE constraint failed')
+        (error.message.includes('UNIQUE constraint failed') ||
+          error.message.includes('Duplicate entry')) // MySQL failure
```

Comment:
could this be changed to instead call the `isDatabaseConflictError` we updated above? I know that maybe not all engines are relevant, but it'd be one less place with duplicated code

Reply:
Good point. I'll switch that out.